tc/core/tc_executor_new_api-inl.h

```diff
@@ -0,0 +1,148 @@
+      if (shapeA[i] != expected.shape[i]) {
+        throw lang::ErrorReport(dbg)
+            << "expected size " << expected.shape[i] << " for dim " << i
+            << " but found " << shapeA[i];
+      }
+    }
+  }
+}
+} // namespace
+
+template <typename Backend>
+TcExecutor<Backend>::TcExecutor(
+    const std::vector<TensorInfo>& inputsInfo,
+    const std::vector<TensorInfo>& outputsInfo,
+    const tc2halide::HalideComponents& halideComponents,
+    const typename Backend::CompilationResultType& compilationResult)
+    : compiledSource(compilationResult.source),
+      inputsInfo_(inputsInfo),
+      outputsInfo_(outputsInfo),
+      halideComponents_(halideComponents),
+      // Parameters (currently) depend on compileWithTcMapper but are
+      // not backend-specific, so we store them in the executor
+      // TODO: revisit this later once we have strides and parametric kernels
+      // with more legitimate uses of parameters.
+      parameters_(compilationResult.parameters) {}
+
+template <typename Backend>
+void TcExecutor<Backend>::run(
+    const std::vector<const DLConstTensor*>& inputs,
+    const std::vector<const DLTensor*>& outputs) const {
+  checkSizesAndStridesAreCompliant(
+      inputs, inputsInfo_, halideComponents_.getDef().params());
+  checkSizesAndStridesAreCompliant(
+      outputs, outputsInfo_, halideComponents_.getDef().returns());
+  std::vector<const void*> I;
+  std::vector<void*> O;
+  for (size_t i = 0; i < inputs.size(); ++i) {
+    I.push_back(inputs[i]->data);
+  }
+  for (size_t i = 0; i < outputs.size(); ++i) {
+    O.push_back(outputs[i]->data);
+  }
+  // Static dispatch instead of virtual functions requires this cast.
+  static_cast<const typename Backend::ExecutorType&>(*this).uncheckedRun(I, O);
+}
+
+template <typename Backend>
+ProfilingInfo TcExecutor<Backend>::profile(
+    const std::vector<const DLConstTensor*>& inputs,
+    const std::vector<const DLTensor*>& outputs) const {
+  auto start = std::chrono::system_clock::now();
+  checkSizesAndStridesAreCompliant(
+      inputs, inputsInfo_, halideComponents_.getDef().params());
+  checkSizesAndStridesAreCompliant(
+      outputs, outputsInfo_, halideComponents_.getDef().returns());
+
+  std::vector<const void*> I;
+  std::vector<void*> O;
+  for (size_t i = 0; i < inputs.size(); ++i) {
+    I.push_back(inputs[i]->data);
+  }
+  for (size_t i = 0; i < outputs.size(); ++i) {
+    O.push_back(outputs[i]->data);
+  }
+
+  // Launch kernel and get **the kernel** time (without CPU overhead)
+  ProfilingInfo pi(
+      // Static dispatch instead of virtual functions requires this cast.
+      static_cast<const typename Backend::ExecutorType&>(*this)
+          .profileUnchecked(I, O));
+
+  // The total CPU overhead is the total time minus the (synchronized) kernel
+  // runtime
+  auto end = std::chrono::system_clock::now();
+  Duration cpuOverhead(end - start);
+  cpuOverhead = cpuOverhead - pi.kernelRuntime;
```

Comment:
Isn't `cpuOverhead` already computed by `CudaTcExecutor::profileUnchecked`?

Reply:
It is but it is also missing the extra CPU overhead from this function.